ui/common/src/wakeLock.ts

```diff
@@ -0,0 +1,40 @@
+const supported: boolean = 'wakeLock' in navigator;
+let shouldLock: boolean = false;
+let sentinel: WakeLockSentinel;
+
+export const request = () => {
+  if (supported) {
+    navigator.wakeLock
+      .request('screen')
+      .then((response: WakeLockSentinel) => {
+        shouldLock = true;
+        sentinel = response;
+      })
+      .catch((error: Error) => {
+        console.error('wakeLock - request failure. ' + error.message);
+      });
+  }
+};
+
+export const release = () => {
+  if (supported) {
+    sentinel
```

Comment:
this will throw an exception if `release` is called before `request`, or before the `navigator.wakeLock.request('screen')` promise completes.